### PR TITLE
Version 2.5.0

### PR DESCRIPTION
**Pull request by joeldenning**
Features:
- `loadRootComponent` option. Gives you the ability to lazy load your rootComponent during the bootstrap lifecycle.
- `<Parcel />` react component, available via `import Parcel from 'single-spa-react/parcel`.
- An `update` lifecycle, provided by default, that applies to parcels
- `SingleSpaContext`, a react@16.3 (when available) context that gives you the single-spa props wherever you need them

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # single-spa-react
 
-Generic lifecycle hooks for React applications that are registered as [applications](https://github.com/CanopyTax/single-spa/blob/master/docs/applications.md#registered-applications) of [single-spa](https://github.com/CanopyTax/single-spa).
+Generic lifecycle hooks for React applications that are registered as either [single-spa applications](https://github.com/CanopyTax/single-spa/blob/master/docs/applications.md#registered-applications) of [single-spa](https://github.com/CanopyTax/single-spa) parcels.
```
Comment (kentmclean) [inline README.md]: `of` should be `or`?

File: src/parcel.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ * To avoid duplicate bundling of react, we do not do this inside of single-spa-react.js.
+ * We also do not set up the prop types in this file to avoid requiring the user of the library
+ * to have prop-types installed and in their browser bundle, since not everyone uses prop types.
+ */
+import React from 'react'
+import {SingleSpaContext} from '../lib/single-spa-react.js'
+
+export default class Parcel extends React.Component {
+  static defaultProps = {
+    wrapWith: 'div'
```
Comment (kentmclean) [inline src/parcel.js]: this is the default, correct?

File: README.md (hunk 1 above)
Comment (joeldenning) [reply README.md]: 👍

File: src/parcel.js (hunk 2 above)
Comment (joeldenning) [reply src/parcel.js]: Yep, you can change it by passing in the `wrapWith` prop as `span` or something else

File: src/parcel.proptypes.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      update: PropTypes.func,
+      name: PropTypes.string,
+    }).isRequired,
+
+    // a loading function
+    PropTypes.func,
+  ]),
+
+  // Optional
+  mountParcel: PropTypes.func.isRequired, // Required if it is not available from SingleSpaContext (i.e., when parcel is not rendered by a single-spa application)
```
Comment (kentmclean) [inline src/parcel.proptypes.js]: I like seeing your adoption of PropTypes. Makes me want to do it more in my own projects

**Pull request merged**